src/core/financial.jl

```diff
@@ -134,8 +208,243 @@ struct Financial
+        1995 => 1.016989,
+        1996 => 1.024362,
+        1997 => 1.034171,
+        1998 => 1.038842,
+        1999 => 1.042804,
+        2000 => 1.038542,
+        2001 => 1.043834,
+        2002 => 1.049992,
+        2003 => 1.056232,
+        2004 => 1.062572,
+        2005 => 1.068587,
+        2006 => 1.074681,
+        2007 => 1.080843,
+        2008 => 1.087068,
+        2009 => 1.093349,
+        2010 => 1.099688,
+        2011 => 1.111515,
+        2012 => 1.122895,
+        2013 => 1.133857,
+        2014 => 1.144425,
+        2015 => 1.154627,
+        2016 => 1.164482,
+        2017 => 1.174010,
+        2018 => 1.183233,
+        2019 => 1.192168,
+        2020 => 1.200834,
+        2021 => 1.209226,
+        2022 => 1.217341,
+        2023 => 1.225191,
+        2024 => 1.232790,
+    )
+    # GDP deflator from BenMAP
+    GDP_deflator = Dict(
+        1980 => 0.478513,
+        1981 => 0.527875,
+        1982 => 0.560395,
+        1983 => 0.578397,
+        1984 => 0.603368,
+        1985 => 0.624855,
+        1986 => 0.636469,
+        1987 => 0.659698,
+        1988 => 0.686992,
+        1989 => 0.720093,
+        1990 => 0.759001,
+        1991 => 0.790941,
+        1992 => 0.814750,
+        1993 => 0.839141,
+        1994 => 0.860627,
+        1995 => 0.885017,
+        1996 => 0.911150,
+        1997 => 0.932056,
+        1998 => 0.946574,
+        1999 => 0.967480,
+        2000 => 1.000000,
+        2001 => 1.028455,
+        2002 => 1.044715,
+        2003 => 1.068525,
+        2004 => 1.096980,
+        2005 => 1.134146,
+        2006 => 1.170732,
+        2007 => 1.204077,
+        2008 => 1.250308,
+        2009 => 1.245860,
+        2010 => 1.266295,
+    )
+
+    if !(stack in ["area", "p150", "p300"])
+        @warn "stack should be one of 'area', 'p150', 'p300'"
+        return false
+    end
+
+    fn_2005 = joinpath(EASIUR_data_lib,"sc_8.6MVSL_$(stack)_pop2005.hdf5")
+    ret_map = JLD.load(fn_2005)
+    if pop_year != 2005
+        fn_growth = joinpath(EASIUR_data_lib,"sc_growth_rate_pop2005_pop2040_$(stack).hdf5")
+        map_rate = JLD.load(fn_growth)
+        for (k,v) in map_rate
+            setindex!(ret_map, v .* (v.^(pop_year - 2005)), k)
+        end
+    end
+    if income_year != 2005
+        try
+            adj = get(MorIncomeGrowthAdj, income_year, nothing) / get(MorIncomeGrowthAdj, 2005, nothing)
+            for (k, v) in ret_map
+                setindex!(ret_map, v .* adj, k)
+            end
+        catch
+            @warn "income year is $(income_year) but must be between 1990 to 2024"
+            return false
+        end
+    end
+    if dollar_year != 2010
+        try
+            adj = get(GDP_deflator, dollar_year, nothing) / get(GDP_deflator, 2010, nothing)
+            for (k, v) in ret_map
+                setindex!(ret_map, v .* adj, k)
+            end
+        catch e
+            @warn "Dollar year must be between 1980 to 2010"
+            return false
```

Comment:
The three `false` returns can be removed because the function is not user accessible, so the conditions which lead to false return will not be met with how the code is written at the moment. I will remove them locally and test them